File: Sort/Intersection of Two Arrays/IntersectionofTwoArrays.cpp
class Solution
{
public:
  vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
  {
    unordered_map<int, bool> map;
    vector<int> result;

    for (int i = 0; i < nums1.size(); i++)
    {
      int num = nums1[i];
      if (map.find(num) == map.end())
      {
        map[num] = false;
      }
    }

    for (int i = 0; i < nums2.size(); i++)
    {
      int num = nums2[i];
      if (map.find(num) != map.end())
      {
        map[num] = true;
      }
    }

    for (unordered_map<int, bool>::iterator iter = map.begin(); iter != map.end(); iter++)
    {
      if (iter->second == true)
      {
        result.push_back(iter->first);
      }
    }
    return result;
  }
};

File: Sort/Intersection of Two Arrays/IntersectionofTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  var obj = {}
  var result = []
  nums1.forEach(v => obj[v] ? null : obj[v] = 1)
  nums2.forEach(v => obj[v] ? obj[v] = 0 : null)
  for (var i in obj) {
    if (obj[i] === 0)
      result.push(+i)
  }
  return result
}